ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/parts/PluginVersionPart.java

```diff
@@ -454,4 +486,43 @@ public void addListeners(ModifyListener minListener, ModifyListener maxListener)
 	protected String getGroupText() {
 		return PDEUIMessages.DependencyPropertiesDialog_groupText;
 	}
+
+	private enum Type {
+        MAJOR, MINOR, MICRO
+    }
+
+    private void bumpVersion(Type type) {
+        String target;
+        try {
+            Version v = new Version(getMinVersion());
+            int maj = v.getMajor(), min = v.getMinor(), mic = v.getMicro();
+            switch (type) {
+                case MAJOR:
+                    maj++;
+                    min = 0;
+                    mic = 0;
+                    break;
+                case MINOR:
+                    min++;
+                    mic = 0;
+                    break;
+                case MICRO:
+                    mic++;
+                    break;
+            }
```

Comment:
Instead defining a new Enum-Type and adding a lot of logic in this method, I think it would be better to pass a `UnaryOperator<Version> versionModifier` instead of `type` and just use that to compute the new version after the current version String was parsed into a `Version` object. That new Version object could then be serialized back into the text field as it's already done now.
With that new approach the caller would pass a lambda that return a corresponding new version object from the current version that was passed to it.